### DISCUSSION TITLE
SO-SOC MISSING, ERROR , Fail to launch module filedatastore

### DISCUSSION THREAD

**Discussion by densnoigaskogen** [Security-Onion-Solutions/securityonion#8628] [Unsupported Versions] [answered]
We have SO running in production for the past years, but it ran into problem with SO-SOC very recently, so-soc status stays **MISSING, ERROR**. Got **401 Authorization Required** when visiting its URL.  I have checked related discussions, but could not fix this issue.

Below are the actions I have taken trying to fix, but without success:
- so-soc-restart 
- sudo docker stop so-soc
   sudo docker rm so-soc
    sudo salt-call state.apply soc
- restart 
- soup (version=2.3.150  )

Can I get some advice on identifying and fix the issue?

 sensoroni-server log shows: 

> timestamp=2022-08-29T20:44:11.64408175Z level=info message="Initializing module" module=staticrbac
timestamp=2022-08-29T20:44:11.644095742Z level=info message=**"Initializing module" module=filedatastore**
timestamp=2022-08-29T20:44:11.660645031Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.660682945Z level=error message="Unable to load job file" error="unexpected end of JSON input" file=jobs/<name-of-sensor-node>/1006.json
timestamp=2022-08-29T20:44:11.660733643Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.660758096Z level=error message="Unable to load job file" error="unexpected end of JSON input" file=jobs/<name-of-sensor-node>/1009.json
timestamp=2022-08-29T20:44:11.660806091Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.660839548Z level=error message="Unable to load job file" error="unexpected end of JSON input" file=jobs/<name-of-sensor-node>/1011.json
timestamp=2022-08-29T20:44:11.660883774Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.660903054Z level=error message="Unable to load job file" error="unexpected end of JSON input" file=jobs/<name-of-sensor-node>/1014.json
timestamp=2022-08-29T20:44:11.660943076Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.660962452Z level=error message="Unable to load job file" error="unexpected end of JSON input" file=jobs/<name-of-sensor-node>/1015.json
timestamp=2022-08-29T20:44:11.661002668Z level=error message="Syntax error reading json object" error="unexpected end of JSON input" offset=0
timestamp=2022-08-29T20:44:11.670479537Z level=error message=**"Failed to initialize module" error="unexpected end of JSON input"  module=filedatastore**
timestamp=2022-08-29T20:44:11.670500501Z level=error message=**"Aborting module launch due to initialization error" error="unexpected end of JSON input"**
timestamp=2022-08-29T20:44:11.671080509Z **level=info message="Terminating Modules"**

![image](https://user-images.githubusercontent.com/33136996/187305602-af3f4020-eb99-4936-997e-c3baf543ec80.png)

**Comment by dougburks** [comment]
When you updated to Elastic 8, did you follow all steps here?
https://docs.securityonion.net/en/2.3/soup.html#elastic-8

**Reply by densnoigaskogen** [reply]
Before upgrading to 2.3.150 , our version was 2.3.140. after running soup on the master node, upgrading went smoothly without any errors.
As for SO-SOC error, we got the same status and log errors before the upgrading. I was hoping that upgrading could fix the issue, since I ran out of ideas, but it did not.

**Reply by dougburks** [reply]
What version were you running before 2.3.140?

When you updated to 2.3.140, did you follow all steps here?
https://docs.securityonion.net/en/2.3/soup.html#elastic-8

**Comment by densnoigaskogen** [comment, marked as answer]
We have fixed the issue by deleting the complaining json files shown in the log, and restart soc. 
/nsm/soc/jobs/xxxx/1006.json
/nsm/soc/jobs/xxxx/1007.json
